File: python/aoc/domain/compute/instructions/math.py
from domain.compute.instructions.instruction import Instruction


class Math(Instruction):
    @property
    def MathOperation(self):
        return self._MathOperation
    @property
    def Length(self):
        return 4
    def __init__(self, opcode, mathOperation):
        super(Math, self).__init__(opcode)
        self._MathOperation = mathOperation

File: Solution/Aoc/Aoc.Domain/Compute/Instructions/Math.cs
using System;

namespace Aoc.Domain.Compute.Instructions
{
    public class Math : Instruction
    {
        public Func<int, int, int> MathOperation { get; }
        public override int Length => 4;

        public Math(int opcode, Func<int, int, int> mathOperation) : base(opcode)
        {
            MathOperation = mathOperation;
            ParameterModes.Add(ParameterMode.Immediate);
        }
    }
}
